### PR TITLE
*: replace JSON with BinaryJSON

### PR DESCRIPTION
**Pull request by coocood**
The serialization and deserialization cost can be avoided.

### CODE REVIEW

**Comment by coocood** [comment]
@zz-jason @jackysp PTAL

**Comment by zz-jason** [comment]
LGTM

File: types/json/binary.go [hunk 1]
```diff
@@ -202,19 +202,23 @@ func (bj BinaryJSON) marshalTo(buf []byte) ([]byte, error) {
...
-func (bj BinaryJSON) getFloat64() float64 {
-	return math.Float64frombits(bj.getUint64())
+// GetFloat64 gets the float64 value.
+func (bj BinaryJSON) GetFloat64() float64 {
+	return math.Float64frombits(bj.GetUint64())
 }
 
-func (bj BinaryJSON) getString() []byte {
+// GetString gets the string value.
+func (bj BinaryJSON) GetString() []byte {
```
Comment (jackysp) [inline types/json/binary.go]: Why not return a string?

File: types/json/binary.go [hunk 2]
```diff
@@ -551,9 +558,15 @@ func appendBinaryNumber(buf []byte, x json.Number) (TypeCode, []byte, error) {
 		typeCode = TypeCodeInt64
 		i64, err := x.Int64()
 		if err != nil {
-			return typeCode, nil, errors.Trace(err)
+			typeCode = TypeCodeFloat64
+			f64, err := x.Float64()
```
Comment (jackysp) [inline types/json/binary.go]: Is this a bug fix?

Comment (coocood) [reply types/json/binary.go]: Yes, to pass the tests in `expression` package.

File: types/json/binary.go (hunk 1 above)
Comment (coocood) [reply types/json/binary.go]: It's more convenient to use []byte internally.

**Pull request merged**